remme/tp/basic.py

```diff
@@ -101,19 +104,22 @@ def apply(self, transaction, context):
                                      format(int(transaction_payload.method)))
         except ParseError:
             raise InvalidTransaction('Cannot decode transaction payload')
-        print(self._family_name)
         addresses = context.set_state({k: v.SerializeToString() for k, v in updated_state.items()})
         if len(addresses) < len(updated_state):
             raise InternalError('Failed to update all of states. Updated: {}. '
                                 'Full list of states to update: {}.'
                                 .format(addresses, updated_state.keys()))
 
-    def make_address(self, appendix):
-        address = self._prefix + appendix
+    @staticmethod
+    def make_address(address):
         if not is_address(address):
             raise InternalError('{} is not a valid address'.format(address))
         return address
 
-    def make_address_from_data(self, data):
+    def make_address_from_data(self, data, prefix=None):
+        # FIXME: bypassing the framework
```

Comment:
With such implementation this don't even look like a dirty hack :)

Reply:
Sarcazm?)